### PR TITLE
adds  filter tests for entity frameworks core 

### PR DESCRIPTION
**Pull request by PascalSenn**
**DEPENDS ON #1688 and  #1736**

Adds integration test for ef core

### CODE REVIEW

File: src/HotChocolate/Filters/test/Types.Filters.SqlLite.Tests/__snapshots__/QueryableFilterTests.Execute_Filter_Not_Equals_Null__sql.snap [hunk 1]
```diff
@@ -0,0 +1,3 @@
+﻿SELECT "d"."Id", "d"."Bar", "d"."Baz", "d"."Qux"
```
Comment (PascalSenn) [inline src/HotChocolate/Filters/test/Types.Filters.SqlLite.Tests/__snapshots__/QueryableFilterTests.Execute_Filter_Not_Equals_Null__sql.snap:1]: Unsued

File: src/HotChocolate/Filters/test/Types.Filters.SqlLite.Tests/__snapshots__/QueryableFilterTests.Execute_Filter_Not_Equals_Null.snap [hunk 2]
```diff
@@ -0,0 +1,27 @@
+﻿{
```
Comment (PascalSenn) [inline src/HotChocolate/Filters/test/Types.Filters.SqlLite.Tests/__snapshots__/QueryableFilterTests.Execute_Filter_Not_Equals_Null.snap:1]: Unsued

File: src/HotChocolate/Filters/test/Types.Filters.SqlLite.Tests/QueryableFilterVisitorEnumTests.cs [hunk 3]
```diff
@@ -0,0 +1,717 @@
...
+                "{ foos(where: { barEnum_in: [QUX, BAZ]) {  totalCount } }");
+
+            // assert
+            var queryResult = (IReadOnlyQueryResult)result;
+
+            Assert.Equal(0, (queryResult.Errors?.Count ?? 0));
+
+
+            result.MatchSnapshot();
+        }
```
Comment (PascalSenn) [inline src/HotChocolate/Filters/test/Types.Filters.SqlLite.Tests/QueryableFilterVisitorEnumTests.cs:690]: Test case for #1704

**Comment by michaelstaib** [comment]
I am closing this one since we changed direction on filters and this one is out of sync.

**Pull request closed without merging**